src/passport-saml/saml.ts

```diff
@@ -955,20 +981,20 @@ class SAML {
     }
   }
 
-  processValidlySignedAssertion(xml, samlResponseXml, inResponseTo, callback) {
+  processValidlySignedAssertion(xml: xml2js.convertableToString, samlResponseXml: string, inResponseTo: string, callback: (err: Error | null, profile?: Profile | undefined, loggedOut?: boolean | undefined) => void) {
     let msg;
     const parserConfig = {
       explicitRoot: true,
       explicitCharkey: true,
       tagNameProcessors: [xml2js.processors.stripPrefix]
     };
     const nowMs = new Date().getTime();
-    const profile: Record<string, any> = {};
-    let assertion;
-    let parsedAssertion;
+    const profile = {} as Profile;
+    let assertion: any;
```

Comment:
I really don't like using `any`. Perhaps a placeholder that is a very generic type could be used here instead of `any`. That still allows flexibility, but also shows that we've though about what is actually happening here. Really, these are objects coming back, so at least we know that much We can extend them later with properties we know as well and then include a generic `[K: string]: any` or something like that. That will help with IntelliSense.